core/src/main/java/org/apache/accumulo/core/client/admin/SummaryRetriever.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Map;
+
+import org.apache.hadoop.io.Text;
+
+public interface SummaryRetriever {
+
+  /**
+   * Allows optionally setting start row before retrieving data.
+   */
+  SummaryRetriever start(Text startRow);
```

Comment:
@joshelser I created something called SummaryStore in ab8deb7.  It generates ~10 summaries for input data without knowing how many keys will be added beforhand.  Uses the approach I mentioned above of merging summaries.  One additional step I discovered I needed to do was double the number of key/vals per summary each time I merged.

So if I have 10 summaries each covering 1000 keys, when I merge to 5 then each of those represents 2000 keys.  So now I want start adding summaries every 2000 keys.  Next time I merge, I will start adding summaries every 4000 keys... etc.